server/routerlicious/packages/services-utils/src/auth.ts

```diff
@@ -3,15 +3,45 @@
  * Licensed under the MIT License.
  */
 
-import { debug } from "util";
 import { ITokenClaims, IUser, ScopeType } from "@fluidframework/protocol-definitions";
 import * as jwt from "jsonwebtoken";
 import { v4 as uuid } from "uuid";
-import { getRandomName } from "./generateNames";
 
 /**
- * Generates a JWT token to authorize routerlicious
+ * Validates a JWT token to authorize routerlicious and returns decoded claims.
+ * An undefined return value indicates invalid claims.
  */
+export function validateTokenClaims(
+    token: string,
+    documentId: string,
+    tenantId: string,
+    maxTokenLifetimeSec: number,
+    isTokenExpiryEnabled: boolean): ITokenClaims {
+    const claims = jwt.decode(token) as ITokenClaims;
+
+    if (!claims || claims.documentId !== documentId || claims.tenantId !== tenantId) {
+        return undefined;
+    }
+
+    if (claims.scopes === undefined || claims.scopes.length === 0) {
+        return undefined;
+    }
+
+    if (isTokenExpiryEnabled && claims.exp && claims.iat) {
+        const now = Math.round((new Date()).getTime() / 1000);
+        if (now >= claims.exp || claims.exp - claims.iat > maxTokenLifetimeSec) {
+            return undefined;
+        }
+    }
+
+    return claims;
+}
+
+/**
+ * Generates a JWT token to authorize routerlicious. This function uses a browser friendly auth library (jsrsasign)
+ * and should only be used in client context.
+ */
+// TODO: We should use this library in all server code rather than using jsonwebtoken directly.
```

Comment:
This is not the browser friendly version. This one is meant to be used inside nodejs only.